components/post_view/post_flag_icon/post_flag_icon.js

```diff
@@ -83,7 +83,7 @@ export default class PostFlagIcon extends React.PureComponent {
                 }
             >
                 <button
-                    id={flagIconId}
+                    id={flagIconId || `${this.props.idPrefix}_${this.props.postId}`}
```

Comment:
Long term ID will be `${this.props.idPrefix}_${this.props.postId}`.  I let the existing `flagIconId` just to make sure that I'm not breaking anything from Se tests.  Will address those IDs separately as it will require major change and verification as well.